File: aws/common/naming.go
// Names for some CloudFormation custom resources.
package common

// SSM Parameters
//
// Some of the custom CloudFormation resources create by the forge are implemented as SSM parameters in the parameter
// store.
const SsmParameterPrefix = "/codesmith-forge"

// Sequence naming
func SequenceParameterName(sequenceName string) string {
	return SsmParameterPrefix + "/Sequence" + sequenceName
}

// CogCondPreAuth naming
func CogCondPreAuthParameterName(userPoolId string, userPoolClientId string) string {
	return SsmParameterPrefix + "/CogCondPreAuth/" + userPoolId + "/" + userPoolClientId
}

File: aws/common/naming.py
# SSM Parameters
#
# Some of the custom CloudFormation resources create by the forge are implemented
# as SSM parameters in the parameter store.
SSM_PARAMETER_PREFIX = '/codesmith-forge'


# CogCondPreAuth naming
def cog_cond_pre_auth_parameter_name(user_pool_id: str, user_pool_client_id: str) -> str:
    return '{0}/CogCondPreAuth/{1}/{2}'.format(SSM_PARAMETER_PREFIX, user_pool_id, user_pool_client_id)


# Sequence naming
def sequence_parameter_name(sequence_name: str) -> str:
    return '{0}/Sequence{1}'.format(SSM_PARAMETER_PREFIX, sequence_name)
